src/pages/core/guides/anchor/how-to-create-a-core-collection.md

```diff
@@ -0,0 +1,440 @@
+
+Some of the accounts in the `CreateAsset` struct are `optional`, meaning their value could be either `Some(account)` or `None`. To handle these optional accounts before passing them to the builder, we use a match statement that allows us to check if an account is present (Some) or absent (None) and based on this check, we bind the account as `Some(account.to_account_info())` if it exists, or as `None` if it doesn't. Like this:
+
+```rust
+let update_authority = match &ctx.accounts.update_authority {
+    Some(update_authority) => Some(update_authority.to_account_info()),
+    None => None,
+};
+```
+
+After preparing all the necessary accounts, we pass them to the `CreateCollectionV2CpiBuilder` and use `.invoke()` to execute the instruction, or `.invoke_signed()` if we need to use signer seeds.
+
+For more details on how the Metaplex CPI Builder works, you can refer to this [documentation](/guides/rust/how-to-cpi-into-a-metaplex-program#using-metaplex-rust-transaction-cpi-builders)
+
+### Additional Actions
+
+Before moving on, What if we want to create the asset with plugins and/or external plugins, such as the `FreezeDelegate` plugin or the `AppData` external plugin, already included? Here's how we can do it.
+
+First, let's add all the additional necessary imports:
+
+```rust
+use mpl_core::types::{
+    Plugin, FreezeDelegate, PluginAuthority,
+    ExternalPluginAdapterInitInfo, AppDataInitInfo, 
+    ExternalPluginAdapterSchema
+};
+```
+
+Then let's create vectors to hold the plugins and external plugin adapters, so we can easily add the plugin (or more) using the right imports!
+
+**Note**: Refer to the [documentation](/core/plugins) if you're not sure on what to use! 
+
+```rust
+let mut plugins: Vec<PluginAuthorityPair> = vec![];
+
+plugins.push(
+  PluginAuthorityPair { 
+      plugin: Plugin::FreezeDelegate(FreezeDelegate {frozen: true}), 
+      authority: Some(PluginAuthority::UpdateAuthority) 
+  }
+);
+
+let mut external_plugin_adapters: Vec<ExternalPluginAdapterInitInfo> = vec![];
+    
+external_plugin_adapters.push(
+  ExternalPluginAdapterInitInfo::AppData(
+    AppDataInitInfo {
+      init_plugin_authority: Some(PluginAuthority::UpdateAuthority),
+      data_authority: PluginAuthority::Address{ address: data_authority },
+      schema: Some(ExternalPluginAdapterSchema::Binary),
+    }
+  )
+);
+```
+
+Lastly, let's integrate these plugins into the `CreateCollectionV2CpiBuilder` program instruction like this:
+
+```rust
+CreateCollectionV2CpiBuilder::new(&ctx.accounts.mpl_core_program.to_account_info())
+  .collection(&ctx.accounts.collection.to_account_info())
+  .payer(&ctx.accounts.payer.to_account_info())
+  .update_authority(update_authority.as_ref())
+  .system_program(&ctx.accounts.system_program.to_account_info())
+  .name(args.name)
+  .uri(args.uri)
+  .plugins(plugins)
+  .external_plugin_adapters(external_plugin_adapters)    
+  .invoke()?;
+```
+
+## Creating the Metadata for the Digital Asset
```

Comment:
Not sure if i would include it here - feels a bit confusing to go from rust to js with not much explanation why? Maybe use a dummy url and just link to one of the upload examples?

Reply:
Agreed, this would be in a program enviroment so upload should have already happened. I feel like the upload part of this is uneeded.

If it was a local Rust script example I could see the point of uploading the image via Rust, but uneeded to for Anchor. Just needs to passed in as a param arg.